### PR TITLE
Changing function call logic in `CodeManager` and `Assembler`

### PR DESCRIPTION
**Pull request by c71n93**
- Remove dispatch for function refs from `CodeManager`
- Add possibility to call functions by name in asm
- Some refactoring (write/read bytes from iostream, other)
- Some asm bugs fixed (adding floating point constants, parsing registers names)

### CODE REVIEW

**Comment by c71n93** [comment]
@levBagryansky check this one, pls

File: src/ChaiVM/interpreter/code-manager/code-manager.cpp [hunk 1]
```diff
@@ -2,16 +2,17 @@
 #include <cassert>
 
 #include "ChaiVM/interpreter/code-manager/code-manager.hpp"
```
Comment (levBagryansky) [inline src/ChaiVM/interpreter/code-manager/code-manager.cpp:4]: Watched until this

Comment (c71n93) [reply src/ChaiVM/interpreter/code-manager/code-manager.cpp:4]: @levBagryansky good job, keep going!

File: src/ChaiVM/interpreter/code-manager/code-manager.cpp [hunk 2]
```diff
@@ -26,48 +27,38 @@ void CodeManager::loadPool(std::istream &istream) {
...
     }
-    Immidiate constant_count;
-    istream.read(reinterpret_cast<char *>(&constant_count),
-                 sizeof constant_count);
-    dispatch_ = std::vector<Immidiate>(constant_count, -1);
+    Immidiate constant_count = readBytes<Immidiate>(istream);
     for (int i = 0; i < constant_count; ++i) {
-        ConstantTag type;
-        istream.read(reinterpret_cast<char *>(&type), sizeof type);
+        ConstantTag type = readBytes<ConstantTag>(istream);
```
Comment (levBagryansky) [inline src/ChaiVM/interpreter/code-manager/code-manager.cpp:32]: @c71n93 nice `readBytes`

File: src/ChaiVM/utils/file-format/chai-file.cpp [hunk 3]
```diff
@@ -78,26 +79,25 @@ ChaiFile::addFunction(chai::interpreter::Immidiate access_flags,
...
-                  sizeof(constants));
+        writeBytes(ofs, &constants);
         for (const std::unique_ptr<Constant> &cnst : pool_) {
             cnst->putType(ofs);
             cnst->write(ofs);
         }
         // since main function too.
-        chai::interpreter::Immidiate funcs = functions_.size() + 1;
-        ofs.write(reinterpret_cast<const char *>(&funcs), sizeof(funcs));
+        chai::interpreter::Immidiate funcs_num = functions_.size() + 1;
```
Comment (levBagryansky) [inline src/ChaiVM/utils/file-format/chai-file.cpp:99]: @c71n93 let's rename it to `funcs`.  or rename `conatants` to `constants_num`.

File: src/frontend/assembler/asmlex.l [hunk 4]
```diff
@@ -10,7 +10,7 @@
 WS      [ \n\t\v]+
 COMMENT #.*
 
-INT             [+-]?[1-9][0-9]*
+INT             [+-]?0|([1-9][0-9]*)
```
Comment (levBagryansky) [inline src/frontend/assembler/asmlex.l:13]: @c71n93 strange

**Review by levBagryansky — COMMENTED** [review]
@c71n93 very nice

File: src/frontend/assembler/asmlex.l (hunk 4 above)
Comment (c71n93) [reply src/frontend/assembler/asmlex.l:13]: 0 wouldn't be integer otherwise

**Review by levBagryansky — APPROVED** [review]
@c71n93 LGBT

**Pull request merged**